src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -367,6 +371,11 @@ protected virtual TxReceipt[] ProcessBlock(
         return receipts;
     }
 
+    /// <summary>
+    /// Builds the block context ouf the block and the release spec.
+    /// </summary>
+    protected virtual BlockExecutionContext BuildBlockContext(Block block, IReleaseSpec spec) => new(block.Header, spec);
```

Comment:
Let's also put `IReleaseSpec` in `BlockExecutionContext`, so we won't have to pass `SpecProvider` or `spec` to some places!

Reply:
Sure don't want to block urgent stuff. Just throwing improvement ideas.